### DISCUSSION TITLE
saving webp to jpg gives strange result

### DISCUSSION THREAD

**Discussion by omuleanu** [SixLabors/ImageSharp#2659] [Q&A] [answered]
I tried saving an webp image to jpg using this code:
```
using var fstream = file.OpenReadStream(); // file is IFormFile
var img = Image.Load(fstream);
await img.SaveAsync(filePath, new JpegEncoder { Quality = 50 });
```
and the result is quite strange, here is the result:
![7](https://github.com/SixLabors/ImageSharp/assets/2010805/e9ed9690-dd4c-4823-8b6e-fe48c17b95e8)
the original I've uploaded as zip, because webp is not allowed here
[before.zip](https://github.com/SixLabors/ImageSharp/files/14200293/before.zip)

**Comment by brianpopow** [comment]
@omuleanu this looks like a bug in the webp decoder, my first guess is something is wrong with the alpha channel decoder.

**Comment by brianpopow** [comment, marked as answer]
@omuleanu This is not actually a bug, I was wrong on that: See [Comment](https://github.com/SixLabors/ImageSharp/issues/2660#issuecomment-1933442535), Also: #2660

Use the following to make the background black:
```
image.Mutate(x => { x.BackgroundColor(Color.Black); });
```

Or save to a format which supports alpha channels.
